velox/exec/MergeSource.cpp

```diff
@@ -135,17 +135,19 @@ class MergeExchangeSource : public MergeSource {
   BlockingReason next(RowVectorPtr& data, ContinueFuture* future) override {
     data.reset();
 
-    if (atEnd_) {
+    if (atEnd_ && !currentPage_) {
```

Comment:
Is this a bug fix? Thanks!

Reply:
Yes, this is a fix for pre-existing issue. I notice a test failing intermittently.